org.eclipse.scout.rt.mail/src/main/java/org/eclipse/scout/rt/mail/smtp/SmtpConnectionPool.java

```diff
@@ -320,17 +351,22 @@ protected void releaseConnection(SmtpConnectionPoolEntry poolEntry) {
    *         provided entry.
    */
   protected SmtpConnectionPoolEntry exchangeConnection(SmtpConnectionPoolEntry oldEntry) throws MessagingException {
+    SmtpServerConfig smtpServerConfig = oldEntry.getSmtpServerConfig();
+
     synchronized (m_poolLock) {
+      checkAndThrowIfDestroyed();
 
       m_leasedEntries.remove(oldEntry);
       safeCloseTransport(oldEntry);
 
-      SmtpConnectionPoolEntry newEntry = createNewConnection(oldEntry.getSmtpServerConfig());
-
-      m_idleEntries.remove(newEntry);
```

Comment:
Removing from idle entries wasn't necessary before too I suppose

Reply:
`createNewConnection` was re-written previously new connection was always added to `m_idleEntries`, in this case it was necessary to remove it from there. However as new connection is always immediately used, method is now named `createNewConnectionWithoutPoolLockAndLease(SmtpServerConfig)`, it is now immediately added to leased connections (within `m_poolLock` in the create method itself).